extras/check_code_style.sh

```diff
@@ -0,0 +1,215 @@
+INCREMENTAL=false
+
+ONLY_FILE=
+
+#
+# usage [exit_code bad_option]
+#
+usage ()
+{
+    if test $1 -ne 0 ; then
+        echo "Unknown option: $2"
+    fi
+
+    cat <<EOF
+Usage: check_code_style.sh [options ...]"
+
+Setup of Tooling is explained here: 
+    https://nest.github.io/nest-simulator/coding_guidelines_c++
+
+Options:
+
+    --help               Print program options and exit
+    --incremental        Do analysis one file after another.
+    --file=/path/to/file Perform the static analysis on this file only.
+    --git-start=SHA      Enter the default SHA for git to start the diff
+                         (default=master)
+    --git-end=SHA        Enter the default SHA for git to end the diff
+                         (default=HEAD)
+    --nest-src=/path     The base directory for the NEST sources
+                         (default=. assuming you execute check_code_style.sh 
+                         from the base directory.)
+    --cppcheck=exe       Enter the executable that is used for cppcheck.
+                         (default=cppcheck)
+    --clang-format=exe   Enter the executable that is used for clang-format.
+                         (default=clang-format)
+    --vera++=exe         Enter the executable that is used for vera++.
+                         (default=vera++)
+EOF
+
+    exit $1
+}
+
+#
+# bail_out message
+#
+bail_out ()
+{
+    echo "$1"
+    exit 1
+}
+
+while test $# -gt 0 ; do
+    case "$1" in
+        --help)
+            usage 0
+            ;;
+        --incremental)
+            INCREMENTAL=true
+            ;;
+        --file=*)
+            ONLY_FILE="$( echo "$1" | sed 's/^--file=//' )"
+            if [ ! -f $ONLY_FILE ]; then
+              bail_out "This is not a valid file: $ONLY_FILE"
+            fi
+            ;;
+        --git-start=*)
+            START_SHA="$( echo "$1" | sed 's/^--git-start=//' )"
+            ;;
+        --git-end=*)
+            END_SHA="$( echo "$1" | sed 's/^--git-end=//' )"
+            ;;
+        --nest-src=*)
+            NEST_SRC="$( echo "$1" | sed 's/^--nest-src=//' )"
+            ;;
+        --cppcheck=*)
+            CPPCHECK="$( echo "$1" | sed 's/^--cppcheck=//' )"
+            ;;
+        --vera++=*)
+            VERA="$( echo "$1" | sed 's/^--vera++=//' )"
+            ;;
+        --clang-format=*)
+            CLANG_FORMAT="$( echo "$1" | sed 's/^--clang-format=//' )"
+            ;;
+        *)
+            usage 1 "$1"
+            ;;
+    esac
+    shift
+done
+
+#
+# sed has different syntax for extended regular expressions
+# on different operating systems:
+# BSD: -E
+# other: -r
+#
+EXTENDED_REGEX_PARAM=r
+/bin/sh -c "echo 'hello' | sed -${EXTENDED_REGEX_PARAM} 's/[aeou]/_/g' "  >/dev/null 2>&1 || EXTENDED_REGEX_PARAM=E
+
+echo "Executing check_code_style.sh"
+echo "Tooling:"
+
+# check vera++ is set up appropriately
+$VERA $NEST_SRC/nest/main.cpp >/dev/null 2>&1 || usage 1 "Executable $VERA for vera++ is not working!"
```

Comment:
On my machine, this command fails because Vera doesn't find the default profile. Although a default profile is in place in `/usr/local/lib/vera++/profiles`. Running with `--profile nest` works.

This test will also fail if one is running outside the NEST source root directory with the `--file` option.